tests/samplers/test_seeded_generate.py

```diff
@@ -75,3 +75,8 @@ def test_random_sample_with_seed(
         # verify requests with the same seed match
         assert outputs[1] == outputs[4]
         assert outputs[2] == outputs[5]
+
+        # verify generations within the same parallel sampling group differ
+        for output in outputs:
+            for sub_output_a, sub_output_b in combinations(output, 2):
+                assert sub_output_a != sub_output_b
```

Comment:
would this test be flaky? e.g. when we generate with `n=10`, if two sequences happen to generate the same answer ...